rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/CostAccountingSpec.scala

```diff
@@ -63,4 +78,95 @@ class CostAccountingSpec extends FlatSpec with TripleEqualsSupport {
 
     assertOutOfPhloError(test)
   }
+
+}
+
+class CostAccountingSpec extends FlatSpec with Matchers with PropertyChecks {
+
+  private[this] def evaluateWithCostLog(
+      initialPhlo: Long,
+      contract: String
+  ) = {
+    implicit val errorLog = new ErrorLog[Task]()
+    implicit val costAlg: CostAccounting[Task] =
+      CostAccounting.unsafe[Task](Cost(initialPhlo))
+    implicit val costL                     = costLog[Task]
+    implicit val cost: _cost[Task]         = loggingCost(costAlg, costL)
+    val dbDir                              = Files.createTempDirectory("cost-accounting-spec-")
+    val size                               = 1024L * 1024 * 1024
+    implicit val logF: Log[Task]           = new Log.NOPLog[Task]
+    implicit val metricsEff: Metrics[Task] = new metrics.Metrics.MetricsNOP[Task]
+
+    (for {
+      costsLoggingProgram <- costL.listen({
+                              for {
+                                runtime <- Runtime
+                                            .create[Task, Task.Par](dbDir, size, StoreType.LMDB)
+                                par <- Interpreter[Task].buildNormalizedTerm(contract)
+                                res <- Interpreter[Task]
+                                        .evaluate(runtime, par, Cost(initialPhlo.toLong))
+                                _ <- runtime.close()
+                                _ <- Task.eval(dbDir.recursivelyDelete())
+                              } yield (res)
+                            }.attempt)
+      (result, costLog) = costsLoggingProgram
+      res               <- errorLog.readAndClearErrorVector
+      _                 = Task.now(res should be(Vector.empty))
+    } yield ((result, costLog))).unsafeRunSync
+  }
+
+  case class ContractWithCost(contract: String, cost: Long)
+
+  val shortslow = ContractWithCost(
+    """new loop in {
+         contract loop(@n) = {
+           match n {
+             0 => Nil
+             _ => loop!(n-1)
+           }
+         } |
+         loop!(10)
+       }""".stripMargin,
+    1766
+  )
+
+  "Total cost of evaluation" should "be equal to the sum of all costs in the log" ignore {
+    val initialPhlo       = 10000L
+    val expectedTotalCost = 1766L
+    val (result, costLog) = evaluateWithCostLog(initialPhlo, shortslow.contract)
+    result shouldBe Right(EvaluateResult(Cost(expectedTotalCost), Vector.empty))
+    costLog.map(_.value).toList.sum shouldEqual expectedTotalCost
+  }
+
+  "Running out of phlogistons" should "stop the evaluation upon cost depletion in a single execution branch" in {
+    val initialPhlo       = 1L
+    val contract          = "@1!(1)"
+    val expectedCosts     = List(Cost(4, "substitution"))
+    val (result, costLog) = evaluateWithCostLog(initialPhlo, contract)
+    result shouldBe Left(OutOfPhlogistonsError)
+    costLog.toList should contain theSameElementsAs (expectedCosts)
```

Comment:
I'd much prefer we'd just use `assert()`. the `should` dsl is way too verbose for my likes.